SAMtemplates/load_balancer/main.yaml

```diff
@@ -0,0 +1,146 @@
+
+Resources:
+  # Generate a new ACM certificate
+  GenerateCertificate:
+    Type: AWS::CertificateManager::Certificate
+    Properties:
+      ValidationMethod: DNS
+      DomainName:
+        Fn::Join:
+          - "."
+          - - !Ref StackName
+            - !ImportValue eps-route53-resources:EPS-domain
+      DomainValidationOptions:
+        - DomainName:
+            Fn::Join:
+              - "."
+              - - !Ref StackName
+                - !ImportValue eps-route53-resources:EPS-domain
+          HostedZoneId: !ImportValue eps-route53-resources:EPS-ZoneID
+
+  # Create a new Application Load Balancer
+  LoadBalancer:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      LoadBalancerAttributes:
+        - Key: idle_timeout.timeout_seconds
+          Value: 60
+      Scheme: internet-facing
+      SecurityGroups:
+        - !ImportValue vpc-resources:VpcDefaultSecurityGroup
+      Subnets:
+        - !Ref SubnetA
+        - !Ref SubnetB
+        - !Ref SubnetC
+
+  # Create Trust Store for mTLS
+  Truststore:
+    Type: AWS::ElasticLoadBalancingV2::TrustStore
+    Condition: CreateTruststore
+    Properties:
+      CaCertificatesBundleS3Bucket: "ci-resources-truststorebucket-1emm0xo65wil5"
+      CaCertificatesBundleS3Key: "fhirfacade-truststore.pem"
+      Name: !Sub "${StackName}-trst"
+
+  # Create Listener with optional mTLS Verify Mode
+  ListenerHTTPS:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    Properties:
+      DefaultActions:
+        - Type: forward
+          TargetGroupArn: !Ref TargetGroup
+      LoadBalancerArn: !Ref LoadBalancer
+      Port: 443
+      Protocol: HTTPS
+      Certificates:
+        - CertificateArn: !Ref GenerateCertificate
+      SslPolicy: ELBSecurityPolicy-TLS13-1-2-2021-06
+      MutualAuthentication:
+        !If
+          - CreateTruststore
+          - {
+              Mode: verify,
+              TrustStoreArn: !GetAtt Truststore.TrustStoreArn
+            }
+          - !Ref "AWS::NoValue"
+
+  # Create Target Group
+  TargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      HealthCheckIntervalSeconds: 10
+      HealthCheckPath: /
+      HealthCheckTimeoutSeconds: 5
+      UnhealthyThresholdCount: 2
+      HealthyThresholdCount: 2
+      Port: 80
+      Protocol: HTTP
+      TargetGroupAttributes:
+        - Key: deregistration_delay.timeout_seconds
+          Value: 60
+      TargetType: ip
+      VpcId: !Ref VPC
+
+  # DNS Record for Load Balancer
+  DNSRecord:
+    Type: AWS::Route53::RecordSet
+    Properties:
+      HostedZoneId: !ImportValue eps-route53-resources:EPS-ZoneID
+      Name: !Join
+        - "."
+        - - !Ref StackName
+          - !ImportValue eps-route53-resources:EPS-domain
+      Type: A
+      AliasTarget:
+        DNSName: !GetAtt LoadBalancer.DNSName
+        EvaluateTargetHealth: False
+        HostedZoneId: !GetAtt LoadBalancer.CanonicalHostedZoneID
+
+Outputs:
+  Endpoint:
```

Comment:
I'd also expect outputs for the load balancer arn, security group ID's, & target group arns,